design-library/src/components/BccPagination/BccPagination.vue

```diff
@@ -0,0 +1,143 @@
+<script setup lang="ts">
+import { ArrowBackIosIcon } from "@bcc-code/icons-vue";
+import { ArrowForwardIosIcon } from "@bcc-code/icons-vue";
+import { toRefs, ref, computed, watchEffect } from "vue";
+import type { Item } from "../BccTable/BccTable.vue";
+import BccSelect from "../BccSelect/BccSelect.vue";
+import BccFormLabel from "../BccFormLabel/BccFormLabel.vue";
+
+type Props = {
+  tableItems: Item[];
```

Comment:
For discussion: I'm not sure if `tableItems` needs to be part of this component. Instead, I feel like we should build pagination into the table component, so that that is the only component that is concerned about the actual data (this doesn't have to happen necessarily in this PR though).

It's still good to have this as a separate component, but instead I feel like it should only concern itself with the actual pagination. That way it can be used for other usecases as well, not just together with our table component. Instead of passing a full `tableItems` array, we could maybe just pass the amount of items so the amount of pages can be correctly calculated?

Curious to hear your opinion on this!

Reply:
i do agree that it should not only be used for the table, the change I've did is that it supports array with any types.

Reply:
@laurensgroeneveld  i have made a alternative version of this using Radix vue libary, it uses passing amount of items instead of a array.
https://github.com/bcc-code/bcc-design/pull/301

Reply:
> @laurensgroeneveld i have made a alternative version of this using Radix vue libary, it uses passing amount of items instead of a array.

Yes, that's what I meant here as well. I'll take a look.

> i do agree that it should not only be used for the table, the change I've did is that it supports array with any types.

I feel like the pagination component should not need to be aware of the actual items at all, but instead just the total amount.